cfme/tests/intelligence/test_reports_with_timelines.py

```diff
@@ -1,16 +1,251 @@
+    policy.assign_events("VM Power Off")
+    policy.assign_actions_to_event("VM Power Off", action)
+    profile = appliance.collections.policy_profiles.create(fauxfactory.gen_alpha(),
+                                                           policies=[policy])
+
+    yield new_vm.assign_policy_profiles(profile.description)
+    if profile.exists:
+        profile.delete()
+    if policy.exists:
+        policy.delete()
+    if action.exists:
+        action.delete()
+
+
+@pytest.fixture
+def report_policy_event_week():
+    if REPORTS_FILE_PATH.exists():
+        yaml_dict = yaml.load(open(str(REPORTS_FILE_PATH), 'r'))['policy_event_week']
+        report = CustomReport(**yaml_dict)
+        report.create()
+        yield report
+        report.delete()
+
+
+@pytest.fixture
+def report_policy_7_days():
+    if REPORTS_FILE_PATH.exists():
+        yaml_dict = yaml.load(open(str(REPORTS_FILE_PATH), 'r'))['policy_event_7_days']
+        report = CustomReport(**yaml_dict)
+        report.create()
+        yield report
+        report.delete()
+
+
+@pytest.fixture
+def report_power_operations():
+    if REPORTS_FILE_PATH.exists():
+        yaml_dict = yaml.load(open(str(REPORTS_FILE_PATH), 'r'))['vm_operations_on_off']
+        report = CustomReport(**yaml_dict)
+        report.create()
+        yield report
+        report.delete()
+
+
+@pytest.fixture
+def report_hosts_data_management():
+    if REPORTS_FILE_PATH.exists():
+        yaml_dict = yaml.load(open(str(REPORTS_FILE_PATH), 'r'))['hosts_data_management']
+        report = CustomReport(**yaml_dict)
+        report.create()
+        yield report
+        report.delete()
+
+
+def test_custom_report_policy_timeline(new_vm, control_policy, report_policy_event_week):
+    logger.info('Will stop VM %r', new_vm.name)
+    wait_for(new_vm.provider.mgmt.stop_vm, [new_vm.name], timeout='7m',
+             message='Stopping VM failed')
+
+    def check_timeline():
+        view = navigate_to(report_policy_event_week, 'Timelines')
+
+        wait_for(lambda: view.is_timeline_displayed,
+                 timeout='4m',
+                 fail_func=lambda: navigate_to(report_policy_event_week, 'Timelines'),
+                 message='timeline not displayed')
+        view.flash.assert_no_error()
+
+        events_list = wait_for(view.chart.get_events, timeout='4m').out
+        logger.info('events_list: %r', str(events_list))
+
+        assert events_list, 'Event list of {} is empty!'.format(report_policy_event_week.title)
+
+        found_events = []
+
+        for evt in events_list:
+            if hasattr(evt, 'target_name'):
+                if (evt.target_name == new_vm.name and evt.assigned_profiles
+                        in new_vm.assigned_policy_profiles):
+                    found_events.append(evt)
+        logger.info('found events: %r', str(found_events))
+
+        return bool(found_events)
+
+    wait_for(check_timeline,
```

Comment:
**REQUIRED**
It seems a bit pointless to me. Maybe we can remove it from here.

Reply:
changed as talked offline